daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -748,7 +748,23 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite)
       _.newInstance(tdmlException.getMessage, tdmlException).asInstanceOf[Exception]
     }
 
-    val exceptionToThrow = junitExceptionInstance.getOrElse(tdmlException)
+    val exceptionToThrow = junitExceptionInstance.getOrElse(
+      // We create a new exception here so that we can propagate the
+      // exception as to why we are not able to construct
+      // an org.junit.AssumptionViolatedException.
+      //
+      // This can be useful if nothing is catching this exception.
+      // The CLI catches this, but won't display the messaging that
+      // elaborates on the cause, because the CLI doesn't utilize junit,
+      // so it expects this to fail.
+      //
+      // But if this isn't caught, and propagates to top level,
+      // then we'll get the diagnostic messaging, including about the
+      // cause.
+      //
+      new TDMLTestNotCompatibleException(testName, implementationName,
+        Option(junitExceptionInstance.failed.get))
```

Comment:
Is it worth adding a CLI test to cover this line? It's not possible to hit using sbt tests since they have junit on the classpath.

Reply:
So a CLI test has to attempt to run a test that has implementation="ibm" let's say, in which case it will fail and will throw this, and the CLI will catch it. I suppose so.